File: repositories/userRepository.js
import { PrismaClient } from '@prisma/client';


const prisma = new PrismaClient();


export const userRepository = {

    async getUserId(usernameReq) {
        return await prisma.users.findUnique({
                where: {
                    username: usernameReq
                },
                select: {
                    id: true
                }
            })
    },

    async addUser(user) {
        return await prisma.users.create({
            data: user,
            select: {
                id: true,
                username: true
            }
        })
    },

    async getUserByUsername(username) {
        return await prisma.users.findUnique({
            where: {
                username: username
            },

            select: {
                id: true,
                username: true,
                password: true
            }
        });
    },

    async updateLastLogin(userId) {
        try {
            return await prisma.users.update({
                where: { id: userId },
                data: { last_login: new Date() }
            });
        } catch (err) {
            console.error('Prisma updateLastLogin error:', err);
            throw err;
        }
    },

    async logout(userId) {
        console.log("Logging out user with ID:", userId);
        return true;
    }
}

File: repositories/user_repository.rb
class UserRepository
  class << self

    def get_user_id(username)
      user = User.find_by(username: username)
      user&.id
    end

    def add_user(attributes)
      user = User.new(attributes)
      if user.save
        { id: user.id, username: user.username}
      else
        raise StandardError.new(user.errors.full_messages.join(", "))
      end
    end

    def login(username)
      user = User.find_by(username: username)
      return nil unless user

      { id: user.id, username: user.username, password: user.password_digest }
    end

    def logout(username)
      Rails.logger.info("Lpgging #{username} with user id: #{user_id} out")
      true
    end
  end
end